qiskit_experiments/data_processing/data_action.py

```diff
@@ -30,46 +30,74 @@ def __init__(self, validate: bool = True):
         self._validate = validate
 
     @abstractmethod
-    def _process(self, datum: Any) -> Any:
+    def _process(self, datum: Any, error: Optional[Any] = None, **options) -> Tuple[Any, Any]:
```

Comment:
Do we need a second argument for error, or can it be part of the first `datum` field?
Also should options be passed to this function, or be part of the class object set at initialization?

Reply:
That advantage of having `**options` is that it lets you specify at run-time some aspects of the data processing, e.g. the `outcome` in a counts dictionary to use.

Reply:
Having the error as a separate field works better with the design. Typically the first node will node receive an `error` but may create one for the next node. The typical example of this is taking single-shot IQ data and return the mean and standard error of the mean to the next node, e.g. a SVD node.

Reply:
Sounds like this needs some more discussion. In the interest of merging the SVD I removed `**options` here: e7884bc